src/gmail.js

```diff
@@ -1089,6 +1142,16 @@ var Gmail = function(localJQuery) {
                 }
             }
         }
+
+        if (threads[0]) {
+            actionType = api.tools.check_event_type(threads[0]);
+        }
+
+        if (actionType) {
+            const new_thread_ids = threads.map(thread => thread[1]);
+            const new_email_ids = threads.map(thread => thread[2][7][3]).reduce((a, b) => a.concat(b), []);
```

Comment:
This code does feel a little "brittle" with regard to how the data is expected to be.

What happens if the structure changes? Will our XHR-intercept code crash and break Gmail? Could this be split into its own method where we are more careful to handle parsing errors?